src/plugins/preauth/pkinit/pkinit_trace.h

```diff
@@ -51,6 +60,27 @@
           "{keyblock}", keyblock)
 #define TRACE_PKINIT_CLIENT_NO_IDENTITY(c)                              \
     TRACE(c, "PKINIT client has no configured identity; giving up")
+#define TRACE_PKINIT_CLIENT_PKCS11_GETSYM_FAILED(c, errstr)             \
+    TRACE(c, "PKINIT client unable to find PKCS11 plugin symbol "       \
+          "C_GetFunctionList: {str}", errstr)
+#define TRACE_PKINIT_CLIENT_PKCS11_FUNC_FAILED(c, func, errstr)         \
+    TRACE(c, "PKINIT client function {str} failed: {str}", func, errstr)
+#define TRACE_PKINIT_CLIENT_PKCS11_NO_MATCH_TOKEN(c)                    \
+    TRACE(c, "PKINIT client PKCS#11 module has no matching tokens")
+#define TRACE_PKINIT_CLIENT_PKCS11_NO_TOKEN(c)                          \
+    TRACE(c, "PKINIT client PKCS#11 module shows no slots with tokens")
+#define TRACE_PKINIT_CLIENT_PKCS11_OPEN(c, name)                        \
+    TRACE(c, "PKINIT client opening PKCS#11 module \"{str}\"", name)
+#define TRACE_PKINIT_CLIENT_PKCS11_OPEN_FAILED(c, errstr)               \
+    TRACE(c, "PKINIT client PKCS#11 module open failed: {str}", errstr)
+#define TRACE_PKINIT_CLIENT_PKCS11_SLOT(c, slot, len, label)            \
+    TRACE(c, "PKINIT client PKCS#11 slotid {int} token {lenstr}",       \
+          slot, len, label)
+#define TRACE_PKINIT_CLIENT_REGEXP_RULE(c, keyword, comp, value)        \
+    TRACE(c, "PKINIT client checking {str} rule '%{str} with value "    \
```

Comment:
The quoting around the second str doesn't look right, and I think it's grown an extra '%'.